storybook/addons/design-system-theme/register.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import { addons, types, useGlobals } from '@storybook/manager-api';
+import { MirrorIcon } from '@storybook/icons';
+import {
+	IconButton,
+	WithTooltip,
+	TooltipMessage,
+	TooltipLinkList,
+} from '@storybook/components';
+
+interface ThemeOption {
+	id: string;
+	title: string;
+}
+
+interface ThemeTooltipMessageProps {
+	title: string;
+	globalName: string;
+	options: ThemeOption[];
+}
+
+const ADDON_ID = '@wordpress/storybook-addon-design-system-theme';
+
+const COLOR_OPTIONS: ThemeOption[] = [
+	{ id: '', title: 'Default' },
+	{ id: 'dark', title: 'Dark' },
+];
+
+const DENSITY_OPTIONS: ThemeOption[] = [
+	{ id: 'compact', title: 'Compact' },
+	{ id: '', title: 'Default' },
+	{ id: 'comfortable', title: 'Comfortable' },
+];
+
+function ThemeTooltipMessage( {
+	title,
+	globalName,
+	options,
+}: ThemeTooltipMessageProps ) {
+	const [ globals, updateGlobals ] = useGlobals();
+	const currentGlobal = globals[ globalName ] ?? '';
+
+	const links = options.map( ( option ) => ( {
+		id: option.id,
+		title: option.title,
+		active: currentGlobal === option.id,
+		onClick: () =>
+			updateGlobals( { [ globalName ]: option.id || undefined } ),
+	} ) );
+
+	return (
+		<TooltipMessage
+			title={ title }
+			desc={ <TooltipLinkList links={ links } /> }
+		/>
+	);
+}
+
+const ThemeTool = () => {
+	return (
+		<WithTooltip
+			placement="top"
+			trigger="click"
+			closeOnOutsideClick
+			tooltip={
+				<>
+					<ThemeTooltipMessage
+						title="Density"
+						globalName="dsDensity"
+						options={ DENSITY_OPTIONS }
+					/>
+					<ThemeTooltipMessage
+						title="Color"
+						globalName="dsColorTheme"
+						options={ COLOR_OPTIONS }
+					/>
+				</>
+			}
+		>
+			<IconButton key="density" title="Design System Density" active>
```

Comment:
```suggestion
			<IconButton title="Design System" active>
```

Also not sure what the `active` prop is here 🤔 And I could not find the documentation for the `@storybook/components` package within a few minutes 🙃

Reply:
> Also not sure what the `active` prop is here 🤔 And I could not find the documentation for the `@storybook/components` package within a few minutes 🙃

Short answer is: `active` makes the toolbar item appear blue instead of a more disabled-looking default gray.

<img width="57" height="34" alt="image" src="https://github.com/user-attachments/assets/7f137216-4099-4e12-89bb-8afa7f365be0" />

As far as documentation... I'm struggling to find it as well 😅 Looking back through my browser history, I found [this Chromatic component library](https://www.chromatic.com/library?appId=5ccbc373887ca40020446347). Today I was also able to track down [the "Storybook Design System" Storybook](https://storybook-design-system.netlify.app/?path=/docs/intro--page) (unclear if official). These include some of the components being used here (e.g. `TooltipMessage`), but not `IconButton` for some reason.

Reply:
Regarding the suggestions, I'll take another look at this. I think some of these may have been sourced from reference documentation ([example](https://storybook.js.org/docs/essentials/toolbars-and-globals#updating-globals-from-within-an-addon)), but it's not entirely clear if they're needed (particularly `key`, not as confident yet on `title`).

Reply:
Went with your suggestion in 80cbbeabf2 , adapted slightly to include "Theme" (as visibly shown).